packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js

```diff
@@ -40,38 +41,38 @@ function SubjectViewer({
   subjectQueueState = asyncStates.initialized,
   subjectReadyState
 }) {
-  const { t } = useTranslation('components')
-  switch (subjectQueueState) {
-    case asyncStates.initialized: {
-      return null
-    }
-    case asyncStates.loading: {
-      return (<div>{t('SubjectViewer.loading')}</div>)
-    }
-    case asyncStates.error: {
-      console.error('There was an error loading the subjects')
-      return null
-    }
-    case asyncStates.success: {
-      const Viewer = getViewer(subject?.viewer)
-      
-      if (Viewer) {
-        return (
-          <Viewer
-            enableInteractionLayer={enableInteractionLayer}
-            key={subject.id}
-            subject={subject}
-            loadingState={subjectReadyState}
-            onError={onError}
-            onReady={onSubjectReady}
-            viewerConfiguration={subject?.viewerConfiguration}
-          />
-        )
-      }
+  const [ViewComponent, setViewComponent] = useState(null);
```

Comment:
Yup, and the convention for naming a component is PascalCase. Because it's a component, I opted for that.

Your StackOverflow isn't accurate because it's focused on rendered components. The way I'm doing it is as a handle on the component's function, so that the component itself is rendered from the state-saved component function declaration. Saving the function declaration in a state variable is the same as storing any function.